### PR TITLE
Add Edge Board Support in Arduino Library Build for TFLu (micro)

### PR DESCRIPTION
**Pull request by oclyke**
Add support for the SparkFun Edge board in the Arduino library build. Allows individuals who are less comfortable with command line tools to get started with TFLu examples. 

**dependencies**
[Update TFLu (micro) to use AmbiqSuite SDK Release 2.2.0 for Apollo3](https://github.com/tensorflow/tensorflow/pull/35236)
[Upgrade Edge Board Support Package in TFLu (micro)](https://github.com/tensorflow/tensorflow/pull/35290)

### CODE REVIEW

**Comment by oclyke** [comment]
Currently the solution for providing the various implementations of data providers and result responders is fairly ugly (using Arduino-provided macros to determine what source code is valid). I will make another attempt to use the Makefile and scripts (python / bash) to get additional source files to be copied over. If anyone with experience with those tools want to chime in I am all ears!

**Comment by oclyke** [comment]
Still having trouble coercing the build to copy additional source files. The desired effect is to make board-specific feature provider and responder files such as:

audio_provider_arduino_nano33ble.cpp
audio_provider_sparkfun_edge.cpp

Each file would still use the preprocessor and board-unique macros from Arduino IDE to exclude the implementation when compiled for any other board.

**Comment by oclyke** [comment]
Latest commits have gotten this branch working well. All three examples in the generated library can be built for either the Arduino Nano 33 BLE or the SparkFun Edge board simply by selecting the correct board in Arduino IDE

**Review by petewarden — DISMISSED** [review]

**Comment by gbaned** [comment]
@oclyke  Can you please check build failures? Thanks!

**Comment by oclyke** [comment]
@gbaned @petewarden 

The Ubuntu Sanity check failed in step 2 (pylint) thanks to missing a space after a comma. I expect that the latest commit will resolve the failure. 

Here's the relevant output that clued me in.


```
9f8df38c-ci_build_gen_ci_sanity_out-shard1run1attempt1-test.log

=== Sanity check step 2 of 15: do_pylint (Python 3 pylint) ===

...

FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/lite/micro/tools/make/fix_arduino_subfolders.py:31: [C0326(bad-whitespace), ] Exactly one space required after comma

...

==== Summary of sanity check results ====
1. do_configure_test: Run ./configure
  [0;32mPASS[0m
2. do_pylint: Python 3 pylint
  [0;31mFAIL[0m
3. do_check_futures_test: Check that python files have certain __future__ imports
  [0;32mPASS[0m
4. do_buildifier: buildifier check
  [0;32mPASS[0m
5. do_bazel_nobuild: bazel nobuild
  [0;32mPASS[0m
6. do_bazel_deps_query: bazel query
  [0;32mPASS[0m
7. do_pip_package_licenses_check: pip: license check for external dependencies
  [0;32mPASS[0m
8. do_lib_package_licenses_check: C library: license check for external dependencies
  [0;32mPASS[0m
9. do_java_package_licenses_check: Java Native Library: license check for external dependencies
  [0;32mPASS[0m
10. do_pip_smoke_test: Pip Smoke Test: Checking py_test dependencies exist in pip package
  [0;32mPASS[0m
11. do_check_load_py_test: Check load py_test: Check that BUILD files with py_test target properly load py_test
  [0;32mPASS[0m
12. do_code_link_check: Code Link Check: Check there are no broken links
  [0;32mPASS[0m
13. do_check_file_name_test: Check file names for cases
  [0;32mPASS[0m
14. do_pip_no_cuda_deps_check_ubuntu: Check Ubuntu gpu pip package does not depend on cuda shared libraries
  [0;32mPASS[0m
15. do_pip_no_cuda_deps_check_windows: Check Windows gpu pip package does not depend on cuda shared libraries
  [0;32mPASS[0m

1 failed; 14 passed.

Sanity checks [0;31mFAILED[0m
```

**Comment by oclyke** [comment]
@gbaned Is there anything I need to do to get the checks to run again? Thanks for your help.

**Comment by gbaned** [comment]
@oclyke Can you please resolve conflicts? Thanks!

**Comment by oclyke** [comment]
@gbaned taken care of, thanks!

**Comment by oclyke** [comment]
@gbaned Is there anything I can do to help move this PR up through the CI queue? We are hoping that we could announce TF Arduino support of the Edge board alongside the 2020 TensorFlow Dev Summit. Thanks!

**Comment by oclyke** [comment]
@gbaned awesome! What are the next steps in your process? Must the ```import/copybara``` check be completed before a merge?

**Comment by gbaned** [comment]
> @gbaned awesome! What are the next steps in your process? Must the `import/copybara` check be completed before a merge?

@oclyke This is waiting for the reviewer approval at this moment. It will be processed further once approved by a reviewer. Thank you!

**Review by petewarden — DISMISSED** [review]

**Comment by oclyke** [comment]
@gbaned will the Windows Bazel and Windows Bazel GPU check failures prevent merging? In my two past PRs I believe they did not delay acceptance. I have also contacted Pete W. and Nat J. together for reviews - since Pete has taken care of it I am not sure that Nat is planning to look. May we remove the 'awaiting review' label?

**Review by rthadur — APPROVED** [review]

File: tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc
Comment (rthadur) [inline tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc:142]: Here is the internal error : `TF_LITE_REPORT_ERROR should be used instead of error_reporter, so log strings can be removed to save space if needed. ` , @oclyke can you please check this ?

File: tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc
Comment (rthadur) [inline tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc:147]: same as above `TF_LITE_REPORT_ERROR should be used instead of error_reporter, so log strings can be removed to save space if needed.`

**Review by rthadur — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc
Comment (oclyke) [reply tensorflow/lite/micro/examples/magic_wand/sparkfun_edge/accelerometer_handler.cc]: I've replaced that usage with the proper macro style and manually tested the changes. Everything is A-ok on this end. Thanks for reviewing.

**Comment by oclyke** [comment]
@googlebot I fixed it.

**Comment by oclyke** [comment]
@rthadur, @gbaned, @petewarden 

I made quite a blunder committing those changes with my personal email - anyhow I fixed the issue with CLAs and the ```TF_LITE_REPORT_ERROR``` macro is now the only way by which the error reporter is accessed. 

I have verified that the project successfully generates the Arduino library and that the source files compile for both the SparkFun Edge and Arduino Nano 33 BLE. 

I think this means we should be good to go, as long as we re-approve and run CI etc... If there is anything more you need from me please say so. Thanks!

**Review by rthadur — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Review by petewarden — APPROVED** [review]

File: tensorflow/lite/micro/examples/person_detection/himax_driver/HM01B0.h
Comment (rthadur) [inline tensorflow/lite/micro/examples/person_detection/himax_driver/HM01B0.h:20]: `Include the directory when naming google .h files `

File: tensorflow/lite/micro/examples/person_detection/himax_driver/HM01B0.h
Comment (rthadur) [inline tensorflow/lite/micro/examples/person_detection/himax_driver/HM01B0.h:25]: `Could not find a newline character at the end of the file.`

**Review by rthadur — CHANGES_REQUESTED** [review]

**Comment by rthadur** [comment]
@oclyke can you please address above review comments ?

**Pull request merged**
